R/legend.R

```diff
@@ -0,0 +1,47 @@
+#' Easily remove legend(s)
+#'
+#' With no argument, will remove all legends. Provide the name(s) of 
+#' specific aesthetic to remove only certain legends.
+#'
+#'
+#' @md
+#' @param ... additional arguments to be passed to
+#'   \code{\link[ggplot2]{element_text}}
+#'
+#' @return either a \code{\link[ggplot2]{theme}} object or a 
+#' \code{\link[ggplot2]{guides}}object, both of which can be used in 
+#' \code{\link[ggplot2]{ggplot2}} calls 
+#' @export
+#' @import rlang
+#' @import ggplot2
+#'
+#' @examples
+#' 
+#' # Remove all legends
+#' ggplot(mtcars, aes(wt, mpg, colour = cyl, size = hp)) +
+#'   geom_point() + easy_remove_legend()
+#' 
+#' # remove just size legend
+#' ggplot(mtcars, aes(wt, mpg, colour = cyl, size = hp)) +
+#'   geom_point() + easy_remove_legend("size")
+#'   
+#' # can also use:
+#' ggplot(mtcars, aes(wt, mpg, colour = cyl, size = hp)) +
+#'   geom_point() + easy_remove_legend(size)
+#'   
+#' # Remove more than one
+#' ggplot(mtcars, aes(wt, mpg, colour = cyl, size = hp)) +
+#'   geom_point() + easy_remove_legend(size, color)
+easy_remove_legend <- function(...){
+
+  vars <- exprs(...)
```

Comment:
Nice `rlang` use! The fact that the NSE/SE aspect doesn't matter makes this super attractive.

I was going to see if we needed to check the input arguments against possible options (e.g. `ggplot2:::.all_aesthetics)` but:

- specifying an aesthetic that isn't used does nothing harmful (`easy_remove_legend(foo)` just does nothing)
- since `guides` itself uses `rename_aes()`, even base names work (`easy_remove_legend(cex)` will remove the `size` guide, and any of `col`, `colour`, `color` will remove that guide.

We should probably note these points in the documentation.

I'm *very* impressed by this function. Great work!